target/linux/mediatek/image/Makefile

```diff
@@ -28,6 +28,7 @@ define Device/Default
   IMAGE/sysupgrade.bin := append-kernel | pad-to 128k | append-rootfs | pad-rootfs | append-metadata
   SUPPORTED_DEVICES := $(subst _,$(comma),$(1))
 ifeq ($(SUBTARGET),mt7623)
+  FILESYSTEMS += ext4
```

Comment:
Could you move this to `Device/bpi_bananapi-r2`? Not all mt7623 devices use ext4.

Reply:
Not quite sure why the Unielec board can't use ext4; it seems almost identical in many respects.

Reply:
Probably Unielec can use ext4, but its image has not been tested with sysupgrade, right? And new nand-based device might be added in the future, which does not rely on ext4.

Reply:
I've ordered a Unielec board so I can test it. It looks like a lot of unification can be done — including using a FIT image instead of a legacy uImage.

I'm less bothered about a hypothetical possibility that there will be future boards which don't have block devices. None of the current setup would work for those anyway.

But I don't care that much about ext4. @lexa2 is right that the upgrade process is a bit suboptimal with a full read/write file system. You still need to explicitly mark the files you want preserved over upgrade in `/lib/keep.d` anyway and you're probably better off with squashfs + f2fs overlay.

Reply:
I've actually made a support patch for a nand-based device and some of it is in my PR. #3066 But you're right, you don't really need to care about the nand stuff here.

If you can test this on the Unielec board then it would be just fine. I'm not familiar with f2fs so I'll just leave it to you.